### ISSUE TITLE
Receiver not registered: com.bumptech.glide.manager.DefaultConnectivityMonitor

### ISSUE THREAD

**Issue by zhuangjiahua** [bumptech/glide#4468]
**Glide Version**:

**Integration libraries**:

**Device/Android Version**:

**Issue details / Repro steps / Use case background**: 

**Glide load line / `GlideModule` (if any) / list Adapter code (if any)**:
```java
Glide.with...
```

**Layout XML**:
```xml
<FrameLayout xmlns:android="...
```

**Stack trace / LogCat**:
```ruby
paste stack trace and/or log here
```

**Comment by sjudd** [comment]
There's explicit checks for registration: https://github.com/bumptech/glide/blob/5acc99f8fc7e772aeef8aa959ff2a9c203ab1aa6/library/src/main/java/com/bumptech/glide/manager/DefaultConnectivityMonitor.java#L72

If you see a logic error there, please let me know.

**Comment by zhuangjiahua** [comment]
Yes, I see this logic! But it still crash Always,  And I don't know why.

**Comment by Gerolmed** [comment]
Ive run into the same issue

**Comment by LiuYX-Skio** [comment]
> Yes, I see this logic! But it still crash Always, And I don't know why.

Hello, has this problem been solved? I also encountered it
